src/details/ArborX_AccessTraits.hpp

```diff
@@ -103,6 +103,26 @@ struct AccessTraitsHelper<AccessTraits<X, Tag>>
   using tag = Kokkos::detected_t<PredicateTagArchetypeAlias, type>;
 };
 
+template <typename X, typename Tag>
+class Adapt
+{
+  X _x;
+  using Access = AccessTraits<X, Tag>;
+
```

Comment:
Probably needs a `static_assert` that tag is a primitives or predicates tag.

Reply:
The idea is that in API public functions like the data structure constructor or the `query` member functions we call first `check_valid_access_traits` and then we "adapt".  So it would be redundant.

Reply:
That's fair. I was thinking using that for internal checking, but maybe it's an overkill.